core/src/main/java/org/infinispan/remoting/transport/jgroups/CommandAwareRpcDispatcher.java

```diff
@@ -52,42 +55,43 @@
  * @author Pedro Ruivo
  * @since 4.0
  */
-public class CommandAwareRpcDispatcher extends RpcDispatcher {
-   public static final RspList<Response> EMPTY_RESPONSES_LIST = new RspList<>();
+public class CommandAwareRpcDispatcher extends MessageDispatcher {
+   private static final RspList<Response> EMPTY_RESPONSES_LIST = new RspList<>();
 
    private static final Log log = LogFactory.getLog(CommandAwareRpcDispatcher.class);
    private static final boolean trace = log.isTraceEnabled();
    private static final boolean FORCE_MCAST = SecurityActions.getBooleanProperty("infinispan.unsafe.force_multicast");
    private static final long STAGGER_DELAY_NANOS = TimeUnit.MILLISECONDS.toNanos(
          SecurityActions.getIntProperty("infinispan.stagger.delay", 5));
-   private static final int REPLY_FLAGS_TO_CLEAR = Message.Flag.RSVP.value() | Message.Flag.SCOPED.value() |
-         Message.Flag.INTERNAL.value() | Message.Flag.NO_TOTAL_ORDER.value();
+   private static final int REPLY_FLAGS_TO_CLEAR =
+         Message.Flag.RSVP.value() | Message.Flag.INTERNAL.value() | Message.Flag.NO_TOTAL_ORDER.value();
    private static final int REPLY_FLAGS_TO_SET = NO_FC.value();
 
    private final InboundInvocationHandler handler;
    private final ScheduledExecutorService timeoutExecutor;
    private final TimeService timeService;
+   private StreamingMarshaller ispnMarshaller;
 
-   public CommandAwareRpcDispatcher(Channel channel, JGroupsTransport transport,
-         InboundInvocationHandler globalHandler, ScheduledExecutorService timeoutExecutor,
-         TimeService timeService) {
-      this.timeService = timeService;
-      this.server_obj = transport;
+   static {
+      ClassConfigurator.addProtocol((short) 1310, CustomRequestCorrelator.class);
```

Comment:
`TEST_PING` ;)

But I think I can use a different `RequestCorrelator` constructor and avoid defining a magic id here.